lib/reek/context/code_context.rb

```diff
@@ -60,12 +62,13 @@ class CodeContext
       #     (send nil :puts
       #       (lvar :x)))
       def initialize(context, exp)
-        @context            = context
         @exp                = exp
         @children           = []
-        @visibility_tracker = VisibilityTracker.new
         @statement_counter  = StatementCounter.new
         @refs               = AST::ObjectRefs.new
+        @visibility         = :public
+
+        @context = context.append_child_context(self) if context
```

Comment:
I have extracted the creation of parent-child relations out of the initializer again.